### ISSUE TITLE
[Android] Entry control ClearButtonVisibility="WhileEditing" tap location incorrect when inside multiple nested Grids

### ISSUE THREAD

**Issue by greg84** [dotnet/maui#14323]
### Description

This could be related: https://github.com/dotnet/maui/issues/12192

If an Entry control is added to two nested Grids, has a fixed width, and ClearButtonVisibility set to WhileEditing, it's not possible to press the clear button. The actual hit area is to the left side of it.

Regression from Xamarin Forms (used to work).

### Steps to Reproduce

Add the following XAML:

````
        <Grid HorizontalOptions="Center">
            <Grid RowDefinitions="Auto, Auto">
                <Label Grid.Row="0" Text="This is a label" />
                
                <Entry Grid.Row="1"
                       ClearButtonVisibility="WhileEditing"
                       WidthRequest="240" />
            </Grid>
        </Grid>
````

1. Run the app on Android.
2. Type something in the text box.
3. Try to press the clear button (it doesn't work).
4. The actual hit area for the button is to the left of the icon.

### Link to public reproduction project repository

See steps to reproduce

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

All

### Did you find any workaround?

Remove the nested grids.

### Relevant log output

```shell
n/a
```

**Comment by andy-schrijvers** [comment]
This doesnt only happen in nested Grids i have the same issue with a entry inside a single vertical stack layout.
The issue does only appear on android, on windows it worked well.

**Comment by PureWeen** [comment]
@greg84 it looks like this one should be fixed in our next SR by the following PR https://github.com/dotnet/maui/pull/12753. 

I'm going to close this issue for now. Once the next SR comes out if you're still seeing the same thing can you open a new issue?

**Comment by greg84** [comment]
@PureWeen I've noticed there's a new version of MAUI available now that should include the fix: https://github.com/dotnet/maui/releases/tag/7.0.81

How do I know what dotnet SDK version contains the new MAUI version? Is there a page somewhere that maps a MAUI version to an SDK version? Maybe this information could be included in the release notes?

Presumably I just wait for VS to prompt me to install the new SDK version, which will include the updated MAUI version?

**Comment by greg84** [comment]
This scenario is fixed in 7.0.81, but another similar scenario is broken. I've made a new issue for it: https://github.com/dotnet/maui/issues/14620
